addons/hr_holidays/models/hr_leave_allocation.py

```diff
@@ -294,66 +298,64 @@ def _compute_can_approve(self):
             else:
                 allocation.can_approve = True
 
-    @api.onchange('number_of_hours_display')
-    def _onchange_number_of_hours_display(self):
+    @api.depends('holiday_type')
+    def _compute_from_holiday_type(self):
+        for allocation in self:
+            if allocation.holiday_type == 'employee':
+                if not allocation.employee_id:
+                    allocation.employee_id = self.env.user.employee_ids[:1].id
```

Comment:
the `.id` isn't needed ;).

Reply:
Even better with `self.env.user.employee_id` ;)